src/scripts/main.js

```diff
@@ -1,3 +1,17 @@
 'use strict';
 
-// write code here
+const tree = document.querySelector('.tree');
+
+tree.addEventListener('click', (e) => {
+  const closestList = e.target.closest('li');
+
+  if (closestList.children.length !== 0) {
```

Comment:
The condition 'closestList.children.length !== 0' checks if the 'closestList' has children, but it doesn't guarantee that the first child is a 'ul' or 'ol' containing list items. This could result in applying styles to unintended elements. Consider checking if the first child is the expected element type.